core/store/src/main/java/org/locationtech/geowave/core/store/cli/type/DescribeTypeCommand.java

```diff
@@ -0,0 +1,108 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.locationtech.geowave.core.cli.annotations.GeowaveOperation;
+import org.locationtech.geowave.core.cli.api.Command;
+import org.locationtech.geowave.core.cli.api.DefaultOperation;
+import org.locationtech.geowave.core.cli.api.OperationParams;
+import org.locationtech.geowave.core.cli.api.ServiceEnabledCommand;
+import org.locationtech.geowave.core.cli.utils.ConsolePrinter;
+import org.locationtech.geowave.core.store.api.DataTypeAdapter;
+import org.locationtech.geowave.core.store.api.QueryBuilder;
+import org.locationtech.geowave.core.store.cli.store.DataStorePluginOptions;
+import org.locationtech.geowave.core.store.cli.store.StoreLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.beust.jcommander.Parameters;
+
+@GeowaveOperation(name = "describe", parentOperation = TypeSection.class)
+@Parameters(commandDescription = "Describes a type with a given name in a data store")
+public class DescribeTypeCommand extends ServiceEnabledCommand<Void> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DescribeTypeCommand.class);
+
+  @Parameter(description = "<store name> <datatype name>")
+  private List<String> parameters = new ArrayList<>();
+
+  private DataStorePluginOptions inputStoreOptions = null;
+
+  /** Return "200 OK" for all describe commands. */
+  @Override
+  public Boolean successStatusIs200() {
+    return true;
+  }
+
+  @Override
+  public void execute(final OperationParams params) {
+    computeResults(params);
+  }
+
+  public List<String> getParameters() {
+    return parameters;
+  }
+
+  public void setParameters(final String storeName, final String adapterId) {
+    parameters = new ArrayList<>();
+    parameters.add(storeName);
+    parameters.add(adapterId);
+  }
+
+  public DataStorePluginOptions getInputStoreOptions() {
+    return inputStoreOptions;
+  }
+
+  @Override
+  public Void computeResults(final OperationParams params) {
+    // Ensure we have all the required arguments
+    if (parameters.size() != 2) {
+      throw new ParameterException("Requires arguments: <store name> <type name>");
+    }
+
+    final String inputStoreName = parameters.get(0);
+    final String typeName = parameters.get(1);
+
+    // Attempt to load store.
+    final File configFile = getGeoWaveConfigFile(params);
+
+    final StoreLoader inputStoreLoader = new StoreLoader(inputStoreName);
+    if (!inputStoreLoader.loadFromConfig(configFile)) {
+      throw new ParameterException("Cannot find store name: " + inputStoreLoader.getStoreName());
+    }
+    inputStoreOptions = inputStoreLoader.getDataStorePlugin();
+
+    LOGGER.info(
+        "Describing everything in store: " + inputStoreName + " with type name: " + typeName);
+    DataTypeAdapter[] types = inputStoreOptions.createDataStore().getTypes();
```

Comment:
There is a `getType` function on DataStore that can get an adapter by type name.  That should eliminate the need for the loop and the check.

Reply:
getType does not yet exist on master. Skipping this change for now.